### PR TITLE
Fix: handle empty bin strings

### PR DESCRIPTION
**Pull request by rhburrows**
**Summary**

This is a followup PR for #6512. It prevents yarn from creating a link to a dependency's root directory within the `.bin/` directory if the dependency has an empty string for "bin" in its package.json.

**Test plan**

After the change:

```sh
rhburrows$ yarn-local add file-loader --cache-folder ./cache
yarn add v1.13.0-0
info No lockfile found.
... OUTPUT ...
✨  Done in 1.17s.
rhburrows$ ls -al node_modules/.bin/
total 0
drwxr-xr-x   3 rhburrows  staff   96 Oct  7 16:59 .
drwxr-xr-x  18 rhburrows  staff  576 Oct  7 16:59 ..
lrwxr-xr-x   1 rhburrows  staff   19 Oct  7 16:59 json5 -> ../json5/lib/cli.js
```

### CODE REVIEW

**Comment by rhburrows** [comment]
Thanks! I wasn't sure if there was a better way to test the change and your patch does a much better job verifying the intent of this PR.

I applied your patch to the branch.

File: __tests__/commands/install/bin-links.js [hunk 1]
```diff
@@ -175,6 +175,15 @@ test('can use link protocol to install a package that would not be found via nod
   });
 });
 
+test('empty bin string does not create a link', (): Promise<void> => {
+  return runInstall({binLinks: true}, 'install-empty-bin', async config => {
+    const binScripts = await fs.walk(path.join(config.cwd, 'node_modules', '.bin'));
+    expect(binScripts).toHaveLength(1);
```
Comment (Gudahtt) [inline __tests__/commands/install/bin-links.js]: This should have length 2 on Windows, as two files are created per binary on Windows (one Bash script for linux-like environments like Cygwin, and one `.cmd` Batch script). This is why the appveyor tests are failing.

**Comment by Gudahtt** [comment]
Great, that seemed to work. The remaining failing test is unrelated to this PR; that test is known to be flaky and will occasionally timeout.

**Comment by rally25rs** [comment]
I kicked off a new appveyor build. Hopefully it passes this time...

**Pull request merged**